### ISSUE TITLE
Split method chain on function contained in nested function call

### ISSUE THREAD

**Issue by munificent** [dart-lang/dart_style#367]
Before:

```dart
      identifier__ = identifier_____.identifier___
          .transform(new StreamTransformer<TypeArg____, Type_>.fromHandlers(
              handleData: (TypeName___ arg__, EventSink<Type_> arg_) {
        ;
      })).asBroadcastStream();
```

After:

```dart
      identifier__ = identifier_____.identifier___
          .transform(new StreamTransformer<TypeArg____, Type_>.fromHandlers(
              handleData: (TypeName___ arg__, EventSink<Type_> arg_) {
        ;
      }))
          .asBroadcastStream();
```

The problem here is that the function argument is nested inside new StreamTransformer.fromHandlers() so it isn't recognized as a block argument. Maybe for method chain splitting we should detect any hard newline within an argument list?

**Comment by munificent** [comment]
Also, before:

```dart
    _trigger.then(ut.expectAsync((result) {
      if (_deferExpectations == null || _deferExpectations == false) {
        body(result);
      } else {
        defer(ut.expectAsync(() => body(result)));
      }
    })).catchError(ut.fail);
```

after:

```dart
    _trigger
        .then(ut.expectAsync((result) {
      if (_deferExpectations == null || _deferExpectations == false) {
        body(result);
      } else {
        defer(ut.expectAsync(() => body(result)));
      }
    }))
        .catchError(ut.fail);
```
